feature-libs/product-configurator/common/components/configurator-cart-entry-bundle-info/configurator-cart-entry-bundle-info.component.spec.ts

```diff
@@ -0,0 +1,350 @@
+  let component: ConfiguratorCartEntryBundleInfoComponent;
+  let fixture: ComponentFixture<ConfiguratorCartEntryBundleInfoComponent>;
+  let htmlElem: HTMLElement;
+  let mockCartItemContext: MockCartItemContext;
+  let commonConfigUtilsService: CommonConfiguratorUtilsService;
+  let configCartEntryBundleInfoService: ConfiguratorCartEntryBundleInfoService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [I18nTestingModule],
+      declarations: [ConfiguratorCartEntryBundleInfoComponent],
+      providers: [
+        { provide: CartItemContext, useClass: MockCartItemContext },
+        {
+          provide: ControlContainer,
+        },
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    commonConfigUtilsService = TestBed.inject(
+      CommonConfiguratorUtilsService as Type<CommonConfiguratorUtilsService>
+    );
+    configCartEntryBundleInfoService = TestBed.inject(
+      ConfiguratorCartEntryBundleInfoService as Type<
+        ConfiguratorCartEntryBundleInfoService
+      >
+    );
+
+    spyOn(
+      commonConfigUtilsService,
+      'isBundleBasedConfigurator'
+    ).and.callThrough();
+    spyOn(
+      configCartEntryBundleInfoService,
+      'retrieveNumberOfLineItems'
+    ).and.callThrough();
+    spyOn(
+      configCartEntryBundleInfoService,
+      'retrieveLineItems'
+    ).and.callThrough();
+
+    fixture = TestBed.createComponent(ConfiguratorCartEntryBundleInfoComponent);
+    component = fixture.componentInstance;
+    htmlElem = fixture.nativeElement;
+    mockCartItemContext = TestBed.inject(CartItemContext) as any;
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should expose orderEntry$', (done) => {
+    const orderEntry: OrderEntry = { orderCode: '123' };
+    component.orderEntry$.pipe(take(1)).subscribe((value) => {
+      expect(value).toBe(orderEntry);
+      done();
+    });
+
+    mockCartItemContext.item$.next(orderEntry);
+  });
+
+  it('should expose quantityControl$', (done) => {
+    const quantityControl = new FormControl();
+    component.quantityControl$.pipe(take(1)).subscribe((value) => {
+      expect(value).toBe(quantityControl);
+      done();
+    });
+
+    mockCartItemContext.quantityControl$.next(quantityControl);
+  });
+
+  it('should expose readonly$', (done) => {
+    component.readonly$.pipe(take(2), toArray()).subscribe((values) => {
+      expect(values).toEqual([true, false]);
+      done();
+    });
+
+    mockCartItemContext.readonly$.next(true);
+    mockCartItemContext.readonly$.next(false);
+  });
+
+  describe('bundle info for cart entry', () => {
+    it('should not be displayed if model provides empty array', () => {
+      mockCartItemContext.item$.next({
+        statusSummaryList: null,
+        configurationInfos: null,
+      });
+      mockCartItemContext.readonly$.next(false);
+
```

Comment:
Add` fixture.detectChanges();` so that the component will rerendered with a new value

Reply:
It was done in the previous commit